[Modules] Format the metadata.json and aligned the Name and Description for all Modules and Updated Readme (#3309)
Azure/ResourceModules: modules/network/network-managers/network-groups/README.md

> # AVNM NetworkGroups `[Microsoft.Network/networkManagers/networkGroups]`

Reviewer: ```suggestion
# Network Manager Network Groups `[Microsoft.Network/networkManagers/networkGroups]`
```

> This module deploys an Azure Virtual Network Manager (AVNM) Network Group.

Reviewer: ```suggestion
This module deploys a Network Manager Network Group.
```
